### PR TITLE
Remove tidycat.

### PR DESCRIPTION
**Pull request by hlinnaka**
System catalogs are now defined like in the upstream, without any special
tidycat headers.

Note: This doesn't change the way pg_proc_gp.h is generated from
pg_proc.sql, by catullus.pl.

This comes with a replacement for generating 4.3.json:
src/backend/catalog/process_foreign_keys.pl. The foreign keys were the
only thing that the json file was used for. AFAIK gpcheckcat is the only
tool that reads that file, and it only paid attention to the foreign key
information.

While working on this, I noticed that a few tables were missing foreign
key declarations. I added FIXME comments on them. Also, a few tables used to
have "vector" references, e.g. pg_index.indclass was an oidvector, where each
element of the array points to pg_class.oid. AFAICS, those declarations were
actually not used for anything. I left those in place as comments, in case we
want to add support in gpcheckcat and the new process_foreign_keys.pl tool
for them, but for now they're just documentation.

This also removes pablopcatso.pl. It read the json file (Ok, so there was
one more user besides gpcheckcat for it), which is now gone. We could create
a tool that reads the same information straight from the header files, or
from a live database, but it was just a developer aid, and I don't think
anyone's used it for quite a while, so I don't think we need a replacement.

### CODE REVIEW

**Comment by hlinnaka** [comment]
@jimmyyih, you've been running gpcheckcat lately. How does this look to you?

**Comment by vraghavan78** [comment]
@ashwinstar @foyzur @d can you please take a look that PR and help merging it. Thanks!

File: src/include/catalog/README.tidycat [hunk 1]
```diff
@@ -20,18 +16,8 @@ to introduce subtle bugs when updating these structures manually.
 The Greenplum tidycat tools eliminate much of the difficulty
```
Comment (jmcatamney) [inline src/include/catalog/README.tidycat]: This reference to tidycat should be removed as well.

File: src/include/catalog/gp_persistent.h [hunk 2]
```diff
@@ -37,35 +37,6 @@
...
-   persistent_serial_num                              bigint    ,
-   previous_free_tid                                  tid       
-   );
-
-   TIDYCAT_ENDFAKEDEF
-*/
-
 #define GpPersistentRelationNodeRelationId 5090
 
 // UNDONE: Change reserved column to relation_bufpool_kind in tidycat above and CATALOG
```
Comment (jmcatamney) [inline src/include/catalog/gp_persistent.h]: Another tidycat reference to be removed.

**Comment by jimmyyih** [comment]
Looks okay to me.

**Comment by hlinnaka** [comment]
Updated the readme, fixed those obsolete tidycat comments, and pushed. Thanks for the review!

**Pull request closed without merging**